sdk/src/client/secret/manager.rs

```diff
@@ -0,0 +1,469 @@
+
+    /// Secret manager that's just a placeholder, so it can be provided to an online wallet, but can't be used for
+    /// signing.
+    Placeholder,
+}
+
+#[async_trait]
+impl Generate<ed25519::PublicKey> for SecretManager {
+    type Options = serde_json::Value;
+
+    async fn generate(&self, options: &Self::Options) -> crate::client::Result<ed25519::PublicKey> {
+        match self {
+            #[cfg(feature = "stronghold")]
+            SecretManager::Stronghold(s) => {
+                let options = <StrongholdSecretManager as Generate<ed25519::PublicKey>>::Options::deserialize(options)?;
+                Ok(s.generate(&options).await?)
+            }
+            #[cfg(feature = "ledger_nano")]
+            SecretManager::LedgerNano(l) => {
+                let options = <LedgerSecretManager as Generate<ed25519::PublicKey>>::Options::deserialize(options)?;
+                Ok(l.generate(&options).await?)
+            }
+            SecretManager::Mnemonic(m) => {
+                let options = <MnemonicSecretManager as Generate<ed25519::PublicKey>>::Options::deserialize(options)?;
+                Ok(m.generate(&options).await?)
+            }
+            #[cfg(feature = "private_key_secret_manager")]
+            SecretManager::PrivateKey(p) => Ok(p.generate(&()).await?),
+            SecretManager::Placeholder => Err(Error::PlaceholderSecretManager),
+        }
+    }
+}
+
+#[async_trait]
+impl Generate<Ed25519Address> for SecretManager {
+    type Options = serde_json::Value;
+
+    async fn generate(&self, options: &Self::Options) -> crate::client::Result<Ed25519Address> {
+        let public_key: ed25519::PublicKey = self.generate(options).await?;
+        Ok(Ed25519Address::from_public_key_bytes(public_key.to_bytes()))
+    }
+}
+
+#[async_trait]
+impl Generate<Vec<ed25519::PublicKey>> for SecretManager {
+    type Options = serde_json::Value;
+
+    async fn generate(&self, options: &Self::Options) -> crate::client::Result<Vec<ed25519::PublicKey>> {
+        match self {
+            #[cfg(feature = "stronghold")]
+            SecretManager::Stronghold(s) => {
+                let options =
+                    <StrongholdSecretManager as Generate<Vec<ed25519::PublicKey>>>::Options::deserialize(options)?;
+                Ok(s.generate(&options).await?)
+            }
+            #[cfg(feature = "ledger_nano")]
+            SecretManager::LedgerNano(l) => {
+                let options =
+                    <LedgerSecretManager as Generate<Vec<ed25519::PublicKey>>>::Options::deserialize(options)?;
+                Ok(l.generate(&options).await?)
+            }
+            SecretManager::Mnemonic(m) => {
+                let options =
+                    <MnemonicSecretManager as Generate<Vec<ed25519::PublicKey>>>::Options::deserialize(options)?;
+                Ok(m.generate(&options).await?)
+            }
+            #[cfg(feature = "private_key_secret_manager")]
+            SecretManager::PrivateKey(_) => todo!(),
```

Comment:
TODO

Reply:
So all of these are here because I'm not sure what we want to do if you try to generate multiple keys using a private key secret manager (which can only ever give you one key). Should we give that key, even if it doesn't match the provided config? Or return an error?